src/seedu/addressbook/Main.java

```diff
@@ -105,12 +108,16 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageOperationException {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            ui.showToUser(e.getMessage() + storage + "can't be ReadOnly");
+            throw e;
```

Comment:
No need to rethrow and catch and print an error message again. Decide on which layer you want to handle the error, it should be only one for most except certain very rare cases. 

Over here, perhaps just handling here is enough, no need to rethrow up to `runCommandLoopUntilExitCommand`